src/user-statuses/EmojiInput.js

```diff
@@ -0,0 +1,86 @@
+/* @flow strict-local */
+import React, { useContext, useCallback, useMemo } from 'react';
+import type { Node } from 'react';
+import { Platform } from 'react-native';
+import type { AppNavigationProp } from '../nav/AppNavigator';
+
+import { ThemeContext } from '../styles/theme';
+import Touchable from '../common/Touchable';
+import Emoji from '../emoji/Emoji';
+import { Icon } from '../common/Icons';
+import type { EmojiType } from '../types';
+import { createStyleSheet, BORDER_COLOR } from '../styles';
+
+export type Value = null | {| +code: string, +name: string, +type: EmojiType |};
+
+export type Props = $ReadOnly<{|
+  value: Value,
+  onChangeValue: Value => void,
+
+  /**
+   * Component must be under the stack nav that has the emoji-picker screen
+   *
+   * Pass this down from props or `useNavigation`.
+   */
+  navigation: AppNavigationProp<>,
+
+  /**
+   * Give appropriate right margin
+   */
+  rightMargin?: true,
+|}>;
+
+/**
+ * A controlled input component to let the user choose an emoji.
+ *
+ * When pressed, opens the emoji-picker screen, and populates with the emoji
+ * chosen by the user, if any.
+ *
+ * Designed for harmony with our Input component. If changing the appearance
+ * of this or that component, we should try to keep that harmony.
+ */
+export default function EmojiInput(props: Props): Node {
+  const { value, onChangeValue, navigation, rightMargin } = props;
+
+  const { color } = useContext(ThemeContext);
+
+  const handlePress = useCallback(() => {
+    navigation.push('emoji-picker', { onPressEmoji: onChangeValue });
+  }, [navigation, onChangeValue]);
+
+  const styles = useMemo(
+    () =>
+      createStyleSheet({
+        touchable: {
+          // Min touch-target size
+          minWidth: 48,
+          minHeight: 48,
+
+          alignItems: 'center',
+          justifyContent: 'center',
+
+          marginRight: rightMargin ? 4 : undefined,
+
+          ...(Platform.OS === 'ios'
+            ? {
+                borderWidth: 1,
+                borderColor: BORDER_COLOR,
+                borderRadius: 2,
+                padding: 8,
+              }
+            : Object.freeze({})),
```

Comment:
Hmm interesting. Why are the border and padding only on iOS?

Reply:
Ah right, this needs a comment (and should possibly change). It's meant to mimic `Input`, which has

```js
const componentStyles = createStyleSheet({
  input: {
    ...Platform.select({
      ios: {
        borderWidth: 1,
        borderColor: BORDER_COLOR,
        borderRadius: 2,
        padding: 8,
      },
    }),
  },
});
```

Reply:
Ah, got it. IIRC that conditional is there in order to conform to platform conventions. This UI might call for something else, though.